packages/ember-core/src/components/pagination.gts

```diff
@@ -0,0 +1,319 @@
+import { fn } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { isTesting, macroCondition } from '@embroider/macros';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { restartableTask, timeout } from 'ember-concurrency';
+import { formatNumber, t } from 'ember-intl';
+import { TrackedArray } from 'tracked-built-ins';
+
+import { bind } from '../helpers/bind.ts';
+import { classes } from '../helpers/classes.ts';
+import NumberInput from './form/number-input.gts';
+import Select from './form/select.gts';
+
+import type { Optional } from '../index.ts';
+
+export interface Meta {
+  start: string | number;
+  count: string | number;
+  total: string | number;
+}
+
+export interface PaginationSignature {
+  Element: HTMLUListElement;
+  Args: {
+    defaultPageSize?: number;
+    enablePageJump?: boolean;
+    meta?: Meta;
+    pageJumpDebounce?: number;
+    pageSizes?: number[];
+    showDetailedMeta?: boolean;
+
+    onChangePage?: (start: number) => void;
+    onChangePageSize?: (pageSize: number) => void;
+  };
+  Blocks: {
+    default: [];
+  };
+}
+
+interface Page {
+  number?: number;
+  current?: boolean;
+  blank?: boolean;
+}
+
+const defaultPageSize = 25;
+const defaultPageJumpDebounce = 600;
+
+export default class Pagination extends Component<PaginationSignature> {
+  @tracked
+  currentPageNumber = this.currentPage;
+
+  get meta() {
+    const meta = this.args.meta ?? ({} as Meta);
+    const start = parseInt(meta.start as string);
+    const count = parseInt(meta.count as string);
+    const total = parseInt(meta.total as string);
+
+    return {
+      start,
+      count,
+      total,
+    };
+  }
+
+  get defaultPageSize() {
+    return (
+      this.args.pageSizes?.[0] ?? this.args.defaultPageSize ?? defaultPageSize
```

Comment:
Is it appropriate to override the passed-in `defaultPageSize` with the first page size?
I think it is reasonable (Though admittedly less likely) to pass in page sizes of `25,50,100`, but default to `50`.